finagle-core/src/test/scala/com/twitter/finagle/loadbalancer/TrafficDistributorTest.scala

```diff
@@ -470,4 +476,41 @@ class TrafficDistributorTest extends FunSuite {
     Await.ready(client.close())
     intercept[ServiceClosedException] { Await.result(client("x")) }
   }
+
+  test("transition to empty balancer on empty address state")(new Ctx{
+    val init: Activity.State[Set[Address]] = Activity.Pending
+    val dest = Var(init)
+    val dist = newDist(dest)
+
+    dest() = Activity.Ok(Set((1, 2.0),(2, 1.0),(3, 2.0)).map(x => WeightedAddress(Address(x._1), x._2)))
+    val bal0 = Await.result(dist())
+    assert(balancers.size == 2)
+
+    dest() = Activity.Ok(Set.empty[Address])
+    val ex = intercept[Exception] {Await.result(dist())}
```

Comment:
is this a no brokers available exception? can we test for that instead of any Exception? (same in the test below)

Reply:
Yes it is, thanks for pointing that out. I will make the suggested change.